### PR TITLE
Data Logger Activity for Memory Graph

### LINKED ISSUE #323: Implement data logger activity
**Describe the issue**

User will have to browse his device to find data sets. Create a data logger feature similar to PSLab where we store a list of files and import from them

**Would you like to work on the issue?**

Issue for GSoC

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #323 
**Changes**: Data Logger Activity for Memory Graph

**Screenshot/s for the changes**: 

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [ ] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[app-debug.zip](https://github.com/fossasia/neurolab-android/files/3337814/app-debug.zip)

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
```
Comment (CloudyPadmal) [inline app/src/main/AndroidManifest.xml]: Why do we need location?

Comment (jddeep) [reply app/src/main/AndroidManifest.xml]: Ah! Will remove it. We don't need to access GPS. Initially I added it if somehow we needed the perm anywhere. Thanks

File: app/src/main/java/io/neurolab/fragments/MemoryGraphFragment.java [hunk 2]
```diff
@@ -80,6 +83,11 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         View rootView = inflater.inflate(R.layout.fragment_memory_graph, container, false);
         ((MemoryGraphParent) getActivity()).setActionBarTitle(getResources().getString(R.string.mem_graph));
         memGraph = rootView.findViewById(R.id.mem_graph);
+        if (getArguments().getString(LOG_FILE_KEY) != null) {
+            filePath = getArguments().getString(LOG_FILE_KEY);
+            Log.d("File Path received: ", "" + filePath);
```
Comment (jddeep) [inline app/src/main/java/io/neurolab/fragments/MemoryGraphFragment.java]: @CloudyPadmal Have a look here. The filePath received is perfectly fine : `/storage/emulated/0/NeuroLab/abcd.csv`. But while carrying out the `importData` function, it seems to show the FileNotFoundException :thinking: Why do you think that is? Can you check this once on your side? I took care of the perms which are granted. If you find something please tell me, till then I am trying different ways.

Comment (jddeep) [reply app/src/main/java/io/neurolab/fragments/MemoryGraphFragment.java]: The LoggerActivity and other functionality works fine. It is just about the above exception. :confused:

Comment (CloudyPadmal) [reply app/src/main/java/io/neurolab/fragments/MemoryGraphFragment.java]: Let me check

Comment (jddeep) [reply app/src/main/java/io/neurolab/fragments/MemoryGraphFragment.java]: @CloudyPadmal Good news! Have fixed the bug. Will push latest changes.

File: app/build.gradle [hunk 3]
```diff
@@ -62,8 +62,11 @@ dependencies {
     implementation 'org.jogamp.jogl:jogl-all:2.3.2'
     implementation 'de.hdodenhof:circleimageview:3.0.0'
     implementation 'com.github.PhilJay:MPAndroidChart:v3.1.0'
+    implementation 'com.android.support:recyclerview-v7:28.0.0'
```
Comment (CloudyPadmal) [inline app/build.gradle]: Can you clean up these dependencies? Move the version cods to root gradle file as done in from line 44~51 and remove duplicates. Also keep similar libraries together (Eg: android libraries goes with android libraries)

File: app/src/main/java/io/neurolab/fragments/MemoryGraphFragment.java [hunk 4]
```diff
@@ -295,6 +301,15 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
         super.onActivityResult(requestCode, resultCode, data);
     }
 
+    private void importLoggedData(String path) {
+        if (!permission)
+            getRuntimePermissions();
+        toggleMenuItem(menu, isPlaying);
+        progressDialog.show();
+        ParseDataAsync parseDataAsync = new ParseDataAsync(path);
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/fragments/MemoryGraphFragment.java]: Can be one line something like `new ParseDataAsync(path).execute()` right?

File: app/src/main/res/layout/logged_data_item.xml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            android:orientation="horizontal">
+
+            <TextView
+                android:id="@+id/file_name_view"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginBottom="8dp"
+                android:layout_marginLeft="16dp"
+                android:layout_marginStart="16dp"
+                android:layout_marginTop="8dp"
```
Comment (CloudyPadmal) [inline app/src/main/res/layout/logged_data_item.xml]: Hardcoded dimensions. These shouldn't be in your PRs anymore. **Please remember these things**

File: app/src/main/res/menu/utility_menu.xml [hunk 6]
```diff
@@ -27,7 +27,13 @@
         android:id="@+id/info_program"
         android:title="@string/program_info_label"
         android:orderInCategory="400"
-        app:showAsAction="always"
+        app:showAsAction="never"
         android:icon="@drawable/ic_info" />
 
+    <item
+        android:id="@+id/data_logger_menu"
+        android:title="Data Logger"
```
Comment (CloudyPadmal) [inline app/src/main/res/menu/utility_menu.xml]: Hardcoded

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/logged_data_item.xml (hunk 5 above)
Comment (jddeep) [reply app/src/main/res/layout/logged_data_item.xml]: Ah! I believed I directed these. Maybe I didn't push :thinking: Will update asap. thanks

File: app/src/main/res/layout/activity_data_logger.xml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:context=".activities.DataLoggerActivity">
+
+    <TextView
+        android:id="@+id/data_logger_blank_view"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:gravity="center"
+        android:text="No logged Data" />
```
Comment (CloudyPadmal) [inline app/src/main/res/layout/activity_data_logger.xml]: Hardcoded

File: build.gradle [hunk 8]
```diff
@@ -45,4 +45,19 @@ ext {
...
+    jsynLibraryVersion = '16.8.0a'
+    javaOscLibraryVersion = '0.3'
+    jsscLibraryVersion = '2.8.0'
+    jtransformsLibraryVersion = '2.4.0'
+    joglLibraryVersion = '2.3.2'
+    circleImageViewLibraryVersion = '3.0.0'
+    mpandroidchartLibraryVersion = '3.1.0'
+    usbSerialLibraryVersion = '6.0.6'
+    appIntroLibraryVersion = '4.2.3'
+    jfreechartLibraryVersion = '1.0.14'
```
Comment (CloudyPadmal) [inline build.gradle:61]: Are we using this library?

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 8 above)
Comment (jddeep) [reply build.gradle:61]: Yeah in the LongTermAnalyzer class.

**Comment by jddeep** [comment]
@CloudyPadmal Conflicts resolved.

**Comment by mariobehling** [comment]
Looks good, but we need to test this more in detail. So, merging it for now. Thank you!

**Pull request merged**